lib/src/main/kotlin/com/lemonappdev/konsist/core/provider/KoImplementationProviderCore.kt

```diff
@@ -2,9 +2,11 @@ package com.lemonappdev.konsist.core.provider
 
 import com.lemonappdev.konsist.api.provider.KoImplementationProvider
 import org.jetbrains.kotlin.psi.KtFunction
+import org.jetbrains.kotlin.psi.KtProperty
 
 internal interface KoImplementationProviderCore : KoImplementationProvider, KoBaseProviderCore {
-    val ktFunction: KtFunction
+    val ktFunction: KtFunction?
+    val ktProperty: KtProperty?
 
-    override fun hasImplementation(): Boolean = ktFunction.hasBody()
+    override fun hasImplementation(): Boolean = ktFunction?.hasBody() ?: false || ktProperty?.hasBody() ?: false
```

Comment:
Here implementation should be empty. No need for:
```
val ktFunction: KtFunction?
val ktProperty: KtProperty?
```

Instead `KoPropertyDeclarationImpl` and `KoFunctionDeclarationImpl`, should have implementation for `hasImplementation()`